inline-caching/cache-loader/src/main/java/io/vmware/event/ConnectionPool.java

```diff
@@ -0,0 +1,49 @@
+package io.vmware.event;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+
+import org.apache.geode.LogWriter;
+import org.apache.geode.cache.CacheFactory;
+
+public class ConnectionPool {
+  private final HikariDataSource dataSource;
+  private HikariConfig config;
+
+  private static LogWriter log;
+
+  static {
+    log = CacheFactory.getAnyInstance().getDistributedSystem().getLogWriter();
+  }
+
+  private ConnectionPool() {
+    String userName = System.getProperty("postgres.username");
+    String password = System.getProperty("postgres.password");
+    String url = "jdbc:postgresql:postgres";
+
+    log.info("username: "+ userName);
+    log.info("password: "+ password);
+
+    config = new HikariConfig();
+    config.setJdbcUrl(url);
+    config.setUsername(userName);
+    config.setPassword(password);
+    config.addDataSourceProperty("cachePrepStmts", "true");
+    config.addDataSourceProperty("prepStmtCacheSize", "250");
+    config.addDataSourceProperty("prepStmtCacheSqlLimit", "2048");
+
+    dataSource = new HikariDataSource(config);
+  }
+  private static ConnectionPool connectionPool;
+
+  public static ConnectionPool getInstance() {
+    return connectionPool==null?connectionPool = new ConnectionPool() : connectionPool;
+  }
```

Comment:
To make this code a little more thread safe, maybe the instantiation of `ConnectionPool` could be ```private static ConnectionPool connectionPool = new ConnectionPool()```

The the `getInstance()` becomes 
``` public static ConnectionPool getInstance(){
      return connectionPool;
}```

Which is thread safe

Reply:
moved